code/modules/awaymissions/mission_code/ruins/gps.dm

```diff
@@ -0,0 +1,17 @@
+//local gps units to let the ruins be found easier.
+
+/obj/item/gps/ruin
+	name = "navigation console"
+	desc = "A console for navigation in local space, gives off a weak signal that can be picked up if sufficiently close."
+	icon = 'icons/obj/terminals.dmi'
+	icon_state = "gps_console"
+	anchored = TRUE
+	local = TRUE
+	gpstag = "Unknown Signal"
+
+/obj/item/gps/ruin/New()
```

Comment:
This proc is not required. The name is already set in the object define

Reply:
Check gps/New(), it sets the name to "global positioning system ([gpstag])" on top of doing some other things.

Reply:
Good point. Overlooked that one. Mappers can't change the name now though due to this. Not sure how `initial(name)` interacts with mapping variables overrides.

Reply:
Hmm, what if I change generic gps in general so that their initial name is NULL and then the renaming in the New only happens if it is null?

Reply:
that is one way to solve it yes. Another option would be to add a variable or such. But the one you suggested will work